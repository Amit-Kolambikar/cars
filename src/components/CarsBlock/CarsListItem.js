import React from "react";
import { Col, Row } from "react-bootstrap";
import { Link } from 'react-router-dom'

const CarsListItem = (props) => {
  const item = props.data;
  return (
    <Row className="carBlockItem" key={item.stockNumber}>
      <Col sm={2}>
        <img src={item.pictureUrl} className="car-image" alt="auto1" width={100}/>
      </Col>
      <Col sm={10}>
        <h5>{item.manufacturerName}</h5>
        <p style={{ color: "#333", fontSize: 12, marginBottom: 0 }}>
          Stock {item.stockNumber}- {item.mileage.number} {item.mileage.unit} -{" "}
          {item.fuelType} - {item.color}
        </p>
        <Link
          style={{ fontSize: 12 }}
          to={{
            pathname: `/cars/${item.stockNumber}`,
            state: { item }
          }}
        >
          {" "}
          View Details
        </Link>
      </Col>
    </Row>
  );
}

export default CarsListItem;