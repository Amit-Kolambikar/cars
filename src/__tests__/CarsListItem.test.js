import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store"; // Smart components

// Component to be tested
import CarsListItem from "../components/CarsBlock/CarsListItem";

const data = {
  stockNumber: 10126,
  manufacturerName: "Skoda",
  modelName: "Roomster",
  color: "green",
  mileage: {
    number: 135573,
    unit: "km"
  },
  fuelType: "Petrol",
  pictureUrl: "http://localhost:3001/car.svg"
};

describe("<CarsListItem />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(<CarsListItem data={data} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
