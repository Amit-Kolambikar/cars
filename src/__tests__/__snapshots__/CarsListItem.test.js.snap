// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarsListItem /> render() renders the component 1`] = `
<Row
  bsPrefix="row"
  className="carBlockItem"
  noGutters={false}
>
  <Col
    sm={2}
  >
    <img
      alt="auto1"
      className="car-image"
      src="http://localhost:3001/car.svg"
      width={100}
    />
  </Col>
  <Col
    sm={10}
  >
    <h5>
      Skoda
    </h5>
    <p
      style={
        Object {
          "color": "#333",
          "fontSize": 12,
          "marginBottom": 0,
        }
      }
    >
      Stock 
      10126
      - 
      135573
       
      km
       -
       
      Petrol
       - 
      green
    </p>
    <Link
      style={
        Object {
          "fontSize": 12,
        }
      }
      to={
        Object {
          "pathname": "/cars/10126",
          "state": Object {
            "item": Object {
              "color": "green",
              "fuelType": "Petrol",
              "manufacturerName": "Skoda",
              "mileage": Object {
                "number": 135573,
                "unit": "km",
              },
              "modelName": "Roomster",
              "pictureUrl": "http://localhost:3001/car.svg",
              "stockNumber": 10126,
            },
          },
        }
      }
    >
       
      View Details
    </Link>
  </Col>
</Row>
`;
